feature:Trigger form missing "Select Recent Configurations" from airflow 2

rich7420 opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->
closes: #56254 

Adding the dropdown allows users to quickly select and reuse previous manual
DAG run configurations, improving workflow efficiency and reducing
manual input errors.

- This PR has a dropdown component with auto-fill functionality
- Also with tests for backend and frontend

results:
![telegram-cloud-photo-size-5-6273931047833635809-y](https://github.com/user-attachments/assets/bd83c14c-0631-4031-a48b-775911706789)
![telegram-cloud-photo-size-5-6273931047833635810-y](https://github.com/user-attachments/assets/d56fbae3-3dea-46b7-adfc-b1fbe34e2654)
![telegram-cloud-photo-size-5-6273931047833635811-y](https://github.com/user-attachments/assets/4e4fd1ab-7edc-480d-8930-006673544c2e)

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

rich7420 committed:
add a recent configrations drop down list

rich7420 committed:
remove extra files

jason810496 commented on airflow-core/src/airflow/api_fastapi/core_api/routes/public/dag_run.py:
```diff
@@ -93,6 +95,51 @@
+    "/recent-configurations",
+    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
+    dependencies=[Depends(requires_access_dag(method="GET", access_entity=DagAccessEntity.RUN))],
+)
```
`airflow-core/src/airflow/api_fastapi/core_api/routes/ui/dag_run.py` might be a better place for this route.

jason810496 commented on airflow-core/src/airflow/api_fastapi/core_api/datamodels/dag_run.py:
```diff
@@ -182,3 +182,18 @@ class DAGRunsBatchBody(StrictBaseModel):
+class RecentConfigurationsResponse(BaseModel):
+    """Response model for recent DAG run configurations collection."""
+    
+    configurations: list[RecentConfigurationResponse] = Field(description="List of recent configurations")
```
```suggestion
class RecentConfigurationsCollectionResponse(BaseModel):
    """Response model for recent DAG run configurations collection."""
    
    configurations: Iterable[RecentConfigurationResponse]
```

jason810496 commented on airflow-core/src/airflow/api_fastapi/core_api/routes/public/dag_run.py:
```diff
@@ -93,6 +95,51 @@
+            "start_date": result.start_date,
+        })
+    
+    return RecentConfigurationsResponse(configurations=configurations)
```
```suggestion
    configurations = session.scalars(query)
    
    return RecentConfigurationsResponse(configurations=configurations)
```

jason810496 reviewed (commented):
This indeed is a new feature to add! LGTM overall.

rich7420 committed:
refactor: change trigger config UX from dropdown to menu button

- Remove RecentConfigurationsDropdown component
- Add menu option 'Trigger again with this config' to Trigger button
- Pre-fill form with selected DAG Run configuration
- Revert unnecessary API changes for recent configurations
- Update translations for new menu option

jscheffl reviewed (approved):
From visual I like it very much as it is now. LGTM!

jason810496 reviewed (commented):
It seems all the API related part were removed from the PR.

Just to double check that does it mean we don't need those API anymore? Or maybe we push on the wrong branch?

pierrejeambrun reviewed (commented):
Looking good to me.

I'm not sure and would love to hear what others think, but I think we should probably only restore 'parameters' config, and not `logical date` `run_id` and `partition key` which should always remain 'cleared' when opening the form.

Otherwise it will be super common to have `409 conflict` response from the API because people will be trying to create runs with already existing logical date / partition key etc....

rich7420 committed:
follow suggestions for 409 conflict

pierrejeambrun reviewed (commented):
LGTM, just one issue, the prefil config order section is different from the normal trigger form. (ordering  is not consistent)

Normal trigger form:

<img width="1059" height="637" alt="Image" src="https://github.com/user-attachments/assets/11a7cf5c-4c7c-4d18-8d11-c1e27041395c" />

Trigger form from 'trigger again with this config':

<img width="1272" height="756" alt="Image" src="https://github.com/user-attachments/assets/ee4924e1-5fb1-4deb-9859-84ebd1449b75" />


Also this doesn't work for the 'backfill' option that could be a follow up, to pre-propulate the backfill form with the chosen configuration too.

rich7420 committed:
follow suggestions and pick up types

rich7420 committed:
fix the staticcheck error
